presto-hive/src/main/java/io/prestosql/plugin/hive/HivePageSource.java

```diff
@@ -162,7 +163,7 @@ public HivePageSource(
                 HiveType fromType = columnMapping.getBaseTypeCoercionFrom().get().getHiveTypeForDereferences(dereferenceIndices).get();
                 HiveType toType = columnMapping.getHiveColumnHandle().getHiveType();
                 if (!fromType.equals(toType)) {
-                    coercers.add(Optional.of(createCoercer(typeManager, fromType, toType)));
+                    coercers.add(Optional.ofNullable(createCoercer(typeManager, fromType, toType)));
```

Comment:
@findepi we could instead refactor `createCoercer` to return an Optional object, thoughts?